docs/concepts/ipfs-gateway.md

```diff
@@ -3,103 +3,120 @@ title: IPFS Gateway
-:::
-
-When a client request for a CID reaches an IPFS gateway, the gateway first checks whether the CID is cached locally. At this point, one of the following occurs:
+- [Learning more](#learning-more)
 
-- **If the CID is cached locally**, the gateway responds with the content referred to by the CID, and the lifecycle is complete.
-
-- **If the CID is not in the local cache**, the gateway will attempt to retrieve it from the network.
-
-The CID retrieval process is composed of two parts, content discovery / routing and content retrieval:
+## Gateway providers
 
-1. In the **content discovery / routing** step, the gateway will determine <VueCustomTooltip label="An IPFS network peer that can provide data specified by a particular CID upon request." underlined multiline>provider</VueCustomTooltip> location; that is, _where_ the data specified by the CID can be found:
+Regardless of who deploys a gateway and where, any IPFS gateway resolves access to any requested IPFS [content identifier](content-addressing.md).
 
-   - Asking peers that it is directly connected to if they have the data specified by the CID.
-   - Query the DHT for the IDs and network addresses of peers that have the data specified by the CID.
+### Your local gateway
 
-2. Next, the gateway performs **content retrieval**, which can be broken into the following steps:
+Your machine may host a gateway as a local service; e.g., at `localhost:8080`. You have a local gateway service if you installed [IPFS Desktop](https://github.com/ipfs-shipyard/ipfs-desktop#ipfs-desktop) or another form of IPFS node.
 
-   1. The gateway connects to the provider.
-   1. The gateway fetches the CIDs content.
-   1. The gateway streams the content to the client.
+### Public gateways
 
-:::callout
-- Learn more about content discovery, routing, retrieval and the subsystems involved in each part of the process in [How IPFS works](./how-ipfs-works.md).
-- Dive into the technical specifications for gateways in the [IPFS HTTP Gateways specification](https://specs.ipfs.tech/http-gateways/) page.
-:::
+Public ([recursive](#recursive-vs-non-recursive-gateways)) gateways are provided by the IPFS various other organizations, including the [IPFS Foundation](./public-utilities.md#public-ipfs-gateways) as a public utility.
 
-## Gateway providers
- 
-Regardless of who deploys a gateway and where, any IPFS gateway resolves access to any requested IPFS [content identifier](content-addressing.md). Therefore, for best performance, when you need the service of a gateway, you should use the one closest to you.
+For a list of public gateways, see the [IPFS Gateways Checker](https://ipfs.fyi/gateways).
 
-### Your local gateway
+## Gateway types
```

Comment:
I wonder if it makes sense to add a "Gateway providers" type for non-recursive gateways